tests/testthat/test-otel.R

```diff
@@ -0,0 +1,50 @@
+test_that("OpenTelemetry tracing works", {
+  skip_if_not_installed("otelsdk")
+
+  record <- otelsdk::with_otel_record({
+    otel_refresh_tracer("DBI")
+
+    con <- dbConnect(RSQLite::SQLite(), ":memory:")
+    dbWriteTable(con, "mtcars", mtcars)
+    dbGetQuery(con, "SELECT * FROM mtcars")
+    dbGetQuery(
+      con,
+      "SELECT COUNT(*) FROM mtcars WHERE cyl = ?",
+      params = list(1:8)
+    )
+    dbReadTable(con, "mtcars")
+    dbRemoveTable(con, "mtcars")
+    dbDisconnect(con)
+  })
+  # reset tracer after tests
+  otel_refresh_tracer("DBI")
+
+  traces <- record$traces
+
+  expect_length(traces, 10L)
+  expect_equal(traces[[1L]]$name, "dbConnect RSQLite")
```

Comment:
Can we do a snapshot test here? We'd need `local_edition(3)` .

Reply:
Two reasons we're going this route for now:
(i) as the otelsdk implementation is maturing, this is so we're not affected by changes there (snapshot tests are skipped on CRAN so we wouldn't restrict otelsdk at least)
(ii) there are session-specific IDs which we'd have to transform and this seems quite tedious